datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -56,14 +59,40 @@ pub trait GroupColumn: Send + Sync {
     ///
     /// Note that this comparison returns true if both elements are NULL
     fn equal_to(&self, lhs_row: usize, array: &ArrayRef, rhs_row: usize) -> bool;
+
     /// Appends the row at `row` in `array` to this builder
     fn append_val(&mut self, array: &ArrayRef, row: usize);
```

Comment:
Maybe as a follow on we can consider removing `append_val` and `equal_to` and simpl change all codepaths to use the vectorized version